### PR TITLE
Get tests passing on recent 0.7

### PR DESCRIPTION
**Pull request by timholy**
This incorporates #74 (this interface has changed multiple times now, and some of the nightly builds are behind).

### CODE REVIEW

**Comment by timholy** [comment]
The Windows nightly seems unfixable because of https://github.com/JuliaLang/julia/issues/25279.

File: test/runtests.jl [hunk 1]
```diff
@@ -12,6 +12,14 @@ else
     throwing_function(bt) = bt[1]
 end
 
+const rseed = Ref(Base.GLOBAL_RNG)  # to get new random directories (see #24445)
+function randtmp()
+    srand(rseed[])
+    dirname = joinpath(tempdir(), randstring(10))
+    rseed[] = Base.GLOBAL_RNG
```
Comment (rfourquet) [inline test/runtests.jl]: I don't understand the point of this line: isn't rseed[] already equal to Base.GLOBAL_RNG?

File: test/runtests.jl [hunk 2]
```diff
@@ -12,6 +12,14 @@ else
     throwing_function(bt) = bt[1]
 end
 
+const rseed = Ref(Base.GLOBAL_RNG)  # to get new random directories (see #24445)
```
Comment (rfourquet) [inline test/runtests.jl]: Any reason not to use a `RandomDevice` object here? it doesn't need seeding, and is used internally in a call to `srand(::MersenneTwister)`. Actually, `randtmp() = joinpath(tempdir(), randstring(RandomDevice(), 10))` should work (unless called too often and too quickly, as `RandomDevice()` opens a file ("/dev/urandom") which is not freed immediately).

File: test/runtests.jl (hunk 1 above)
Comment (timholy) [reply test/runtests.jl]: I can't remember anymore.

File: test/runtests.jl (hunk 2 above)
Comment (timholy) [reply test/runtests.jl]: Didn't know about RandomDevice, that seems useful.

**Pull request merged**